app/controllers/course/survey/responses_controller.rb

```diff
@@ -28,11 +28,22 @@ def edit
   end
 
   def update
-    @response.submit if params[:response][:submit]
-    if @response.update_attributes(response_update_params)
-      render_response_json
-    else
-      render json: { errors: @response.errors }, status: :bad_request
+    if params[:response][:unsubmit]
```

Comment:
@weiqingtoh Correct me if I'm wrong, but both submit and unsubmit creates patch request which is handled by `update` in the controller. So at best you mean I should refactor `update` such that it calls a method for submit and another for unsubmit?